Source/JavaScriptCore/dfg/DFGFixupPhase.cpp

```diff
@@ -3451,6 +3451,13 @@ class FixupPhase : public Phase {
             return;
         }
 
+        if (node->child1()->shouldSpeculateString()) {
+            fixEdge<StringUse>(node->child1());
+            node->setOpAndDefaultFlags(ToNumber);
+            node->clearFlags(NodeMustGenerate);
+            return;
+        }
```

Comment:
Why do we not need to `setResult` here?

Reply:
We do not need it since `setOpAndDefaultFlags` makes it default one (JS value result for this `ToNumber` opcode).